msal/tests/Test.MSAL.NET.Unit.net45/PlatformProxyTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using Microsoft.Identity.Core;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Test.MSAL.NET.Common.Core.Helpers;
+
+namespace Test.MSAL.NET.Unit.net45
+{
+    [TestClass]
+    public class PlatformProxyTests
+    {
+        [TestMethod]
+        public void ValidateGetDeviceIdPerformance()
+        {
+            var platformProxy = PlatformProxyFactory.GetPlatformProxy();
+            // Call it once to pre-load it.  We're not worried about the time it takes to call it
+            // the first time, we're worried about subsequent calls.
+            platformProxy.GetDeviceId();
+
+            using (new PerformanceValidator(5, "GetDeviceId"))
+            {
+                platformProxy.GetDeviceId();
```

Comment:
>platformProxy.GetDeviceId(); [](start = 16, length = 28)

consider calling all the properties on the platformproxy and broaden the scope of the test a bit

Reply:
fixed